programs/stake/src/stake_instruction.rs

```diff
@@ -4090,6 +4086,14 @@ mod tests {
                     &id(),
                 )
                 .unwrap();
+
+                // some of these tests don't make perfect sense for both Initialized and
+                // Stake, so override the expected result if it's incorrect
+                let expected_result = if split_amount == 0 {
+                    Err(InstructionError::InsufficientFunds)
+                } else {
+                    expected_result
+                };
```

Comment:
I don't feel great about this, and I also didn't feel great about copying the whole test to remove certain cases, which was the first version of this.  Suggestions are welcome!

Reply:
Yeah... I went the make-separate-tests for Initialize and for Stake.

Reply:
Yeah... I went the make-separate-tests for Initialize and for Stake. I think that's what I would vote for here.